pkg/providers/vsphere/vsphere.go

```diff
@@ -493,11 +389,7 @@ func (p *vsphereProvider) SetupAndValidateUpgradeCluster(ctx context.Context, cl
 		return err
 	}
 
-	err := p.setupSSHAuthKeysForUpgrade()
-	if err != nil {
-		return fmt.Errorf("failed setup and validations: %v", err)
-	}
-	err = p.validateMachineConfigsNameUniqueness(ctx, cluster, clusterSpec)
```

Comment:
Shouldn't we also generate ssh keys during upgrade if it's set to empty? Otherwise, there will be a way to upgrade a cluster to not have SSH keys at all, when that's not allowed during create.

Reply:
If you look at what `setupSSHAuthKeysForUpgrade` was doing, it never generated new keys for empty fields. It only stripped the comment from them. That's a difference in behavior between create and upgrade. Not sure why that was done in the first place, but with this refactor I tried to keep as much behavior as possible intact.

Reply:
hmm interesting, we can track this in a separate issue